aten/src/ATen/native/mps/operations/Distributions.mm

```diff
@@ -3,287 +3,204 @@
 #include <ATen/native/Distributions.h>
 #include <ATen/native/DistributionTemplates.h>
 #include <ATen/native/mps/OperationUtils.h>
+#include <ATen/core/PhiloxRNGEngine.h>
 
 namespace at {
 namespace native {
+namespace mps {
 
-Tensor& uniform_mps_(Tensor& input, double from, double to, c10::optional<Generator> gen_)
+struct RandomCachedGraph : public MPSCachedGraph
 {
-  using namespace mps;
-
-  if (input.numel() == 0) {
-    return input;
+  RandomCachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {
```

Comment:
Can we use the OperationUtils.mm API here. It needs to be done from different places. PyTorch has a general API to add Random ops implemented.

Reply:
- RandomCachedGraph is not a generator and is just a 'state maintainer' for each random graph. It has nothing to do with MPSGenerator and cannot be refactored in there. It's per graph, and not global like MPSGenerator.

- MPSGeneratorImpl is a 'Seed generator interface' provided by PyTorch to pass generator parameters from the Python side. Users can pass Seeds manually via that interface (like Python example in below). At this point, we use PyTorch's Default generator to generate seeds from /dev/urandom on MacOS.
```
g_cpu = torch.Generator()
g_cpu.manual_seed(2147483647) 
torch.randint(3, 10, (2, 2), generator=g_cpu)
```